pkg/controllers/state/cluster.go

```diff
@@ -51,11 +51,12 @@ type Cluster struct {
 	cloudProvider cloudprovider.CloudProvider
 	clock         clock.Clock
 
-	mu               sync.RWMutex
-	nodes            map[string]*StateNode           // provider id -> cached node
-	bindings         map[types.NamespacedName]string // pod namespaced named -> node name
-	nameToProviderID map[string]string               // node name -> provider id
-	daemonSetPods    sync.Map                        // daemonSet -> existing pod
+	mu                       sync.RWMutex
+	nodes                    map[string]*StateNode           // provider id -> cached node
+	bindings                 map[types.NamespacedName]string // pod namespaced named -> node name
+	nodeNameToProviderID     map[string]string               // node name -> provider id
```

Comment:
I see we only really need this for the podAntiAffinity function now. It'd be nice if we could just remove this eventually.

Reply:
Sorry, it highlighted a bunch of stuff here. Which part do you think that we should remove?

Reply:
just the nodeNameToProviderID

Reply:
I'm not sure that that's feasible. The real issue here is:
1. With node bindings, we need to know the node mapping to get the state node
2. When we do a delete, we only get the node name (and we don't get the providerID) so we need some way to map this name back to providerID for cleanup